<launch>
  <!-- Arguments -->
  <arg name="cfg_file" default="$(find sgloop_ros)/../../../config/scannet.yaml"/>
  <arg name="output_folder" default="/home/wuxin/Desktop/FM-Fusion/data/ScanNet/output/online_mapping"/>
  <arg name="verbose_level" default="2"/>
  <arg name="frame_gap" default="1"/>
  <arg name="max_frames" default="5000"/>
  <arg name="visualization" default="1"/>
  <arg name="debug" default="false"/>

  <!-- Visualization parameters -->
  <group ns="viz">
    <param name="edge_width" value="0.03"/>
    <param name="edge_color/r" value="0.0"/>
    <param name="edge_color/g" value="1.0"/>
    <param name="edge_color/b" value="1.0"/>
    <param name="centroid_size" value="0.2"/>
    <param name="centroid_v_offset" value="3.0"/>
    <param name="annotation_size" value="0.3"/>
    <param name="annotation_v_offset" value="0.3"/>
  </group>

  <!-- Online Mapping Node -->
  <node pkg="sgloop_ros" name="agentA" type="OnlineMappingNode" output="screen" clear_params="true">
    <param name="cfg_file" value="$(arg cfg_file)"/>
    <param name="output_folder" value="$(arg output_folder)"/>
    <param name="local_agent" value="agentA"/>
    <param name="max_frames" value="$(arg max_frames)"/>
    <param name="frame_gap" value="$(arg frame_gap)"/>
    <param name="visualization" value="$(arg visualization)"/>
    <param name="o3d_verbose_level" value="$(arg verbose_level)"/>
    <param name="debug" value="$(arg debug)"/>
    
    <!-- Topic remapping if needed -->
    <!--
    <remap from="/synced_frame" to="/your_synced_frame_topic"/>
    -->
  </node>

  <!-- Static transform publisher -->
  <node pkg="tf" type="static_transform_publisher" name="link_1st_agent" args="0 0 0 0 0 0 1 world agentA 100" />

  <!-- Optional: RViz for visualization -->
  <!-- 
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sgloop_ros)/rviz/online_mapping.rviz" if="$(arg visualization)"/>
  -->

</launch>
